diag_manager/fms_diag_file_object.F90

```diff
@@ -326,14 +332,13 @@ pure function get_file_unlimdim (this) result(res)
   res = this%diag_yaml_file%get_file_unlimdim()
 end function get_file_unlimdim
 
-!! TODO - get functions for sub region stuff
 !> \brief Returns a copy of file_sub_region from the yaml object
 !! \return Copy of file_sub_region
-!pure function get_file_sub_region (obj) result(res)
-! class(fmsDiagFile_type), intent(in) :: obj !< The file object
-! integer :: res
-!  res = obj%diag_yaml_file%get_file_sub_region()
-!end function get_file_sub_region
+function get_file_sub_region (obj) result(res)
+ class(fmsDiagFile_type), target, intent(in) :: obj !< The file object
```

Comment:
Why is this a `target`?  This function should be pure.

Reply:
```F90
function get_file_sub_region (obj) result(res)
 class(fmsDiagFile_type), intent(in) :: obj !< The file object
 type(subRegion_type) :: res
  res = obj%diag_yaml_file%get_file_sub_region()
end function get_file_sub_region

```

Reply:
Well I learned something here.  I wonder if this requires some more engineering...